server.cpp

```diff
@@ -85,23 +147,33 @@ int main(int argc, char* argv[])
         tcp::resolver resolver(io_service);
         tcp::acceptor acceptor(io_service, *resolver.resolve(tcp::resolver::query(host, port)));
 
+        namespace pls = std::placeholders;
+
+        char buff[1024];
+
         for (;;)
         {
             tcp::socket socket(io_service);
             acceptor.accept(socket);
 
-            const std::string message = make_daytime_string();
+            const size_t bytes = read(socket, boost::asio::buffer(buff), std::bind(read_complete, buff, pls::_1, pls::_2));
 
-            boost::system::error_code ignored_error;
-            boost::asio::write(socket, boost::asio::buffer(message), ignored_error);
+            const std::string msg(buff, bytes);
 
-            std::cout << message << " " << socket.remote_endpoint().address().to_string() << "\n";
+            const size_t str_end_pos = msg.find("\r");
+            const std::string start_str = msg.substr(0, str_end_pos);
+            const std::string http_version = start_str.substr(str_end_pos - 8);
```

Comment:
How about making a class:
```c++
class Request
{
    public:
        Request(const std::string& start_str); // <-- Here goes your start string
        const std::string& verb() const;
        const std::string& path() const;
        const std::string& version() const;
};
```
This class would represent a request: <HTTP-verb> <path> <HTTP-version>

The constructor would split the first request string into three parts: verb, path and version, and store them inside the class.

You can later use an object of such class to check if the version is correct, if the request verb is right and if the path is serviceable.